subxt/src/blocks/extrinsic_types.rs

```diff
@@ -558,6 +607,117 @@ impl<T: Config> ExtrinsicEvents<T> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtensions<'a> {
+    bytes: &'a [u8],
+    metadata: Metadata,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtension<'a> {
+    bytes: &'a [u8],
+    ty_id: u32,
+    identifier: &'a str,
+    metadata: Metadata,
+}
+
+impl<'a> ExtrinsicSignedExtensions<'a> {
+    /// Get a certain signed extension by its name.
+    pub fn find(&self, signed_extension: impl AsRef<str>) -> Option<ExtrinsicSignedExtension> {
+        self.iter()
+            .find_map(|e| e.ok().filter(|e| e.name() == signed_extension.as_ref()))
+    }
+
+    pub fn iter(&'a self) -> impl Iterator<Item = Result<ExtrinsicSignedExtension<'a>, Error>> {
+        let signed_extension_types = self.metadata.extrinsic().signed_extensions();
+        let num_signed_extensions = signed_extension_types.len();
+        let mut index = 0;
+        let mut byte_start_idx = 0;
+
+        std::iter::from_fn(move || {
+            if index == num_signed_extensions {
+                None
+            } else {
+                let extension = &signed_extension_types[index];
+                let ty_id = extension.extra_ty();
+                let cursor = &mut &self.bytes[byte_start_idx..];
+                if let Err(err) = scale_decode::visitor::decode_with_visitor(
+                    cursor,
+                    ty_id,
+                    self.metadata.types(),
+                    scale_decode::visitor::IgnoreVisitor,
+                )
+                .map_err(|e| Error::Decode(e.into()))
+                {
+                    index = num_signed_extensions; // (such that None is returned in next iteration)
```

Comment:
Why not decode all things in the signed_ext? 

Couldn't be useful to get all the things in the signed_ext? Such as you may want to know everything that failed or just filter the ones that were successful?

Reply:
@niklasad1 I could definitely do that, but chances are if one fails to decode, the bytes from here onward are probably not representing what the user thinks they are. I think it is unlikely any following signed extensions are decoded correctly in that case. I was trying to mimic the API of `Extrinsics`, where we have a similar return on error policy in the the `iter()` function.

Reply:
okey, I see uniform API makes sense to me :)